File: server/app/models/session.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base


SESSION_STATUS_STREAMING = 0
SESSION_STATUS_IDLE = 1


class Session(Base):
    __tablename__ = "sessions"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(255), nullable=False, default='')
    agent_id = Column(Integer, nullable=False, index=True)
    status = Column(Integer, default=SESSION_STATUS_IDLE, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    messages = relationship(
        "Message",
        back_populates="session",
        primaryjoin="Session.id == Message.session_id",
        foreign_keys="Message.session_id"
    )

File: server_go/internal/model/session.go
package model

import "time"

const (
	SessionStatusStreaming = 0
	SessionStatusIdle      = 1
)

type Session struct {
	ID        int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	Title     string    `gorm:"type:varchar(255);not null;default:''" json:"title"`
	AgentID   int64     `gorm:"not null;index;column:agent_id" json:"agent_id"`
	Status    int       `gorm:"not null;default:1" json:"status"`
	CreatedAt time.Time `gorm:"not null;autoCreateTime" json:"created_at"`
	UpdatedAt time.Time `gorm:"not null;autoUpdateTime" json:"updated_at"`
}

func (Session) TableName() string { return "sessions" }
